### PR TITLE
ldapsearch (LDAPS + more modular options)

### PR DESCRIPTION
**Pull request by P-aLu**
Added the support of LDAPS and channel binding to the BOF
The "--ldaps" option must be set to use LDAPS and the channel binding support is dynamic.

The "ldapsearch" alias from SA.cna has been changed in order to make it more modular with options.

The "cACertificate" is now treated as an LDAP property that is B64 encoded in result.

Related to https://github.com/Tw1sm/pyldapsearch/pull/2 and https://github.com/coffeegist/bofhound/pull/8

### CODE REVIEW

**Comment by Tw1sm** [comment]
@Palu08 I came here to see how the base64 encoding of the `cACertificate` attribute is handled if the attribute contains more than one cert (was going to match output formatting in pyldapsearch), but this PR appears bugged  



Same result if I manually supply all the arg values with the modified flag names

Not sure how the list scenario will get handled here, but if the attribute contained a list, instead of just 1 binary cert, I base64 and comma separated b64 blobs in  [pyldapsearch](https://github.com/Tw1sm/pyldapsearch/blob/main/pyldapsearch/__main__.py#L371)

**Comment by Palu08** [comment]
Hello @Tw1sm 

Thank you for your review.

I might have patched the CNA to resolve your problem. Added 2 new functions to parse arguments, they may be used on other function than ldapsearch, what do you think ?

Let me know if this did well fix the use case u showed :)

File: SA/SA.cna [hunk 1]
```diff
@@ -175,18 +180,88 @@ alias ldapsearch {
...
+        $arg = @args[$i];
+		$option = split(' ', $arg)[0];
+		$value = substr($arg, strlen($option));
+		$value = replaceQuotes($value);
+
+        if ($option iswm "attributes") {
+            $attributes = $value;
+        }
+        else if ($option iswm "count") {
+            $count = $value;
```
Comment (freefirex) [inline SA/SA.cna:217]: We reference result_limit in what we actually send, meaning count would never be respected

**Review by freefirex — APPROVED** [review]
Just a couple issues I'll hotfix, This is a nice change. I do like how you handled the challenging aspect of attempting to take optional values with flags, one of the first implementations I've seen that didn't break on me.

:100:

**Pull request merged**